dbt/include/bigquery/macros/materializations/table.sql

```diff
@@ -113,10 +113,19 @@ else:
   msg = f"{type(df)} is not a supported type for dbt Python materialization"
   raise Exception(msg)
 
+# For writeMethod we need to use "indirect" if materializing a partitioned table
+# otherwise we can use "direct". Note that indirect will fail if the GCS bucket has a retention policy set on it.
+{%- if partition_config is not none %}
```

Comment:
nit: I'd flip this because it just reads easier to have the user not balance a negative in their head